File: backend/PetCare/Entities/PetProfile.cs
using PetCare.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace PetCare.Entities;

public class PetProfile
{
    [Key]
    public int Id { get; set; }

    [ForeignKey(nameof(Pet))]
    public int PetId { get; set; }

    [JsonIgnore] // 🔥 để tránh vòng lặp ngược lại
    public Pet Pet { get; set; } = null!;

    [MaxLength(255)]
    public string? Personality { get; set; }

    [MaxLength(255)]
    public string? FavoriteFood { get; set; }

    [MaxLength(255)]
    public string? Hobbies { get; set; }

    [MaxLength(255)]
    public string? Story { get; set; }

    [MaxLength(255)]
    public string? SocialImage { get; set; }
}

File: frontend/petcare/lib/models/pet_profile.dart
class PetProfile {
  final int? id;
  final int? petId;
  final String? personality;
  final String? favoriteFood;
  final String? hobbies;
  final String? story;
  final String? socialImage;

  const PetProfile({
    this.id,
    this.petId,
    this.personality,
    this.favoriteFood,
    this.hobbies,
    this.story,
    this.socialImage,
  });

  factory PetProfile.fromJson(Map<String, dynamic> json) {
    return PetProfile(
      id: json['id'] as int?,
      petId: json['petId'] as int?,
      personality: json['personality'] as String?,
      favoriteFood: json['favoriteFood'] as String?,
      hobbies: json['hobbies'] as String?,
      story: json['story'] as String?,
      socialImage: json['socialImage'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'petId': petId,
      'personality': personality,
      'favoriteFood': favoriteFood,
      'hobbies': hobbies,
      'story': story,
      'socialImage': socialImage,
    };
  }
}


